app/src/main/java/cz/muni/fi/pv256/movio2/uco_410434/ui/FilmListFragment.java

```diff
@@ -56,31 +54,68 @@ public void onCreate(@Nullable Bundle savedInstanceState) {
     @Override
     public void onStart() {
         super.onStart();
+        MainThreadBus.getInstance().register(this);
+
 
         if (FilmManager.getInstance().hasData()) {
             fillFilmsList();
             fillListView();
         } else {
-            if (asyncTask != null) {
-                asyncTask.cancel(true);
+            Intent intent = new Intent(getActivity(), FilmsDownloadService.class);
+            getActivity().startService(intent);
+        }
+        Log.i(TAG, "onStart");
+    }
+
+
+    @Subscribe
+    public void onCompleted(FilmsDownloadService.CompletedEvent e) {
+        Log.i(TAG, "completedEvent");
+        Runnable r = new Runnable() {
+            @Override
+            public void run() {
+                fillFilmsList();
+                fillListView();
             }
+        };
 
-            asyncTask = new FilmsLoadAsyncTask(this);
-            asyncTask.execute();
+        if (Looper.myLooper() == Looper.getMainLooper()) {
+            r.run();
+        } else {
+            mHandler.post(r);
```

Comment:
toto je nejake komplikovane. preco sa nemoze volat handler vzdy? preco vobec toto osetrenie?

Reply:
Ahoj, toto som si všimol v task8 a mám to tam opravené.